src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -88,7 +88,8 @@ export class Kernel implements IKernel {
     private readonly kernelExecution: KernelExecution;
     private startCancellation = new CancellationTokenSource();
     constructor(
-        public readonly uri: Uri,
+        public readonly notebookUri: Uri,
+        public readonly resourceUri: Uri,
```

Comment:
Today in our notebook provider we have the distinction between IdentityUri & ResourceUri.
This change merely ensures we support that in the IKernel object as well.